From 386a332946b72f58793f9a4c72859cb9ff12d4c6 Mon Sep 17 00:00:00 2001
From: Philippe Gerum <rpm@xenomai.org>
Date: Sat, 15 Feb 2020 11:14:26 +0100
Subject: [PATCH] Revert "ARM: dovetail: notify co-kernel about und_usr"

This reverts commit 5e6273e9d9c23c4e7c5fa0feab07b594fd669663.

und_usr traps are properly handled directly from the entry code
regardless of the current execution stage. The companion core has no
business in receiving them.

This fixes an ugly bug for the ARM architecture causing fpu-intensive
tasks running out-of-band to repeatedly trigger exception reports by
EVL.
---
 arch/arm/kernel/entry-armv.S | 12 ------------
 1 file changed, 12 deletions(-)

diff --git a/arch/arm/kernel/entry-armv.S b/arch/arm/kernel/entry-armv.S
index 67517f26f3af..19936cb1235a 100644
--- a/arch/arm/kernel/entry-armv.S
+++ b/arch/arm/kernel/entry-armv.S
@@ -496,18 +496,6 @@ ENDPROC(__irq_usr)
 __und_usr:
 	usr_entry uaccess=0
 
-#ifdef CONFIG_DOVETAIL
-	get_thread_info tsk
-	ldr	r0, [tsk, #TI_PREEMPT]		@ get preempt count
-	tst	r0, #TI_OOB_MASK		@ oob stage?
-	beq	1f
-	mov	r0, #ARM_TRAP_UNDEFINSTR
-	mov	r1, sp				@ r1 = &regs
-	bl	__oob_trap_notify
-	uaccess_enable ip
-1:
-#endif
-
 	mov	r2, r4
 	mov	r3, r5
 
-- 
2.16.4

